src/client/src/rt-platforms/glue/glue.ts

```diff
@@ -0,0 +1,96 @@
+import Glue, { Glue42 as GlueInterface } from '@glue42/desktop'
+import Glue4Office, { Glue42Office as Glue42OfficeInterface } from '@glue42/office'
+import { CurrencyPairPositionWithPrice } from 'rt-types'
+import { WindowConfig } from '../types'
+import { BasePlatformAdapter } from '../platformAdapter'
+import { openGlueWindow, registerWindowMethods } from './window'
+import { sendNotification, NotifyPermission } from './utils/sendNotification'
+import { getAddinStatus, openSheet, updateSheet, BlotterData } from './utils/excel'
+import DefaultRoute from 'rt-platforms/defaultRoute'
+import { ApplicationEpic } from 'StoreTypes'
+import { GlueHeader, GlueLogoLink } from './'
+
+// TODO remove onGlueLoaded and its usages when the whole app is rendered after glue has loaded
+export const onGlueLoaded = (callback: () => void) => {
+  const interval = setInterval(() => {
+    if (window.glue) {
+      callback()
+      clearInterval(interval)
+    }
+  }, 100)
+}
+
+/**
+ * Glue implementation of the base platform adapter.
+ * Glue4Office is an optional library for working with excel.
+ */
+export class Glue42 extends BasePlatformAdapter {
+  allowTearOff = true
+  readonly name = 'glue'
+  readonly type = 'desktop'
+
+  constructor() {
+    super()
+    Glue({ channels: true })
+      .then((glue: GlueInterface.Glue) => {
+        window.glue = glue
+        registerWindowMethods()
+        return Glue4Office()
+      })
+      .then((glue4office: Glue42OfficeInterface.Glue4Office) => {
+        window.glue4office = glue4office
```

Comment:
this could be done outside of the promise

Reply:
If my understanding is correct, loading Glue4Office happens after Glue has loaded and this is where we should asynchronously render the application, once the refactoring is done